lib/internal/fs/streams.js

```diff
@@ -100,98 +67,156 @@ const FileHandleOperations = (handle) => {
   };
 };
 
-function close(stream, err, cb) {
-  if (!stream.fd) {
-    // TODO(ronag)
-    // stream.closed = true;
-    cb(err);
-  } else {
-    stream[kFs].close(stream.fd, (er) => {
-      stream.closed = true;
-      cb(er || err);
-    });
-    stream.fd = null;
+function FileSystemStream(path, options) {
```

Comment:
This doesn't need to be a constructor function. Can be just a regular function taking stream as first param.

Reply:
But it's taking the `options` to, well _construct_ the `(Read|Write)Stream` instance (defining properties on it)…